### ISSUE TITLE
Metabolic Syndrome Prediction | 5. Model prediction and evaluation

### ISSUE THREAD

**Issue by SrijanShovit** [SrijanShovit/HealthLearning#98]
### Is your feature request related to a problem? Please describe.

After all data play, let's delve into actual training part.

### Describe the solution you'd like

1. Write generic functions.
2. Use all features; no feature engineering (transformation, selection, extraction, dropping here)
3. Use both normalized and standardized data.
4. Use hyperparameter tuning for models.
5. Use some training strategy like LOSO or K-Fold.
6. Record time to find best params and time to train using those params.
7. Predict on test data and print all eval metrics.
8. If it is binary classification, plot ROC to find optimal threshold.
9. Using that, plot Confusion matrix to evaluate the models.
10. Models to use: Extra Trees, Extra Tree, SVM, Logistic, KNN, DT, RF, GB, Bagging Clf, XGB, Hist GB, MLP, Catboost, Adaboost, Naive Bayes, LightGBM

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SrijanShovit** [comment]
@Arihant-Bhandari proceed

**Comment by Arihant-Bhandari** [comment]
hi @SrijanShovit due to certain circumstances in my college, i am now able to complete this project, i would be really glad if you could open this issue up again. thank you as always

**Comment by Arihant-Bhandari** [comment]
Thank you @SrijanShovit will complete this soon.

**Comment by Taranpreet10451** [comment]
Please assign this to me.

**Comment by Ar7109** [comment]
@SrijanShovit I have worked earlier in machine learning field. Please assign to me under gssoc 2024
